service/storage_proxy.cc

```diff
@@ -1726,11 +1726,17 @@ class storage_proxy::cancellable_write_handlers_list : public bi::list<abstract_
         // handler is being removed from the b::list, so if any live iterator points at it,
         // move it to the next object (this requires that the list is traversed in the forward
         // direction).
```

Comment:
The whole thing is odd. What does it mean to hold a vector of pointers to iterators?

Reply:
Ah, I see it's to protect it from preemption.

Reply:
Good work by our debug mode.

Reply:
> The whole thing is odd. What does it mean to hold a vector of pointers to iterators?

You don't say! Tell it to the author of the code: 301246f6c019b9f93d0d553a347f1a09329429ed

Reply:
> Good work by our debug mode.

The thing I did not manage to figure out is why it started to trigger only now.

Reply:
> > The whole thing is odd. What does it mean to hold a vector of pointers to iterators?
> 
> You don't say! Tell it to the author of the code: [301246f](https://github.com/scylladb/scylladb/commit/301246f6c019b9f93d0d553a347f1a09329429ed)

By "odd" I meant "creative".

Reply:
> > Good work by our debug mode.
> 
> The thing I did not manage to figure out is why it started to trigger only now.

Hmm. Maybe @wmitros can guess, we changed some view update management recently.

Reply:
> > > The whole thing is odd. What does it mean to hold a vector of pointers to iterators?
> > 
> > 
> > You don't say! Tell it to the author of the code: [301246f](https://github.com/scylladb/scylladb/commit/301246f6c019b9f93d0d553a347f1a09329429ed)
> 
> By "odd" I meant "creative".

Don't complement the author too much. He will blush.

Reply:
> > > Good work by our debug mode.
> > 
> > 
> > The thing I did not manage to figure out is why it started to trigger only now.
> 
> Hmm. Maybe @wmitros can guess, we changed some view update management recently.

The only related mv change that I can think of is https://github.com/scylladb/scylladb/pull/19586, which isn't really new, but it add a new place for removing write response handlers which could increment the live iterators to `end()` and cause the issue with dereferencing IIUC
However, that patch is only used when there's no live replicas for the write which shouldn't happen in https://github.com/scylladb/scylladb/issues/20874 - and that test doesn't even use materialized views

Reply:
> You don't say! Tell it to the author of the code: [301246f](https://github.com/scylladb/scylladb/commit/301246f6c019b9f93d0d553a347f1a09329429ed)

Avi: using too many obscure abbreviations is confusing and makes code hard to understand
also Avi: `_vuhl`

:stuck_out_tongue: